contrib/sly-package-fu.el

```diff
@@ -317,4 +332,126 @@ symbol in the Lisp image if possible."
+           (package-name (sly-format-symbol-for-defpackage package)))
+       (insert "(:import-from )")
+       (backward-char)
+       (insert package-name)
+       (newline-and-indent)
+       (insert symbol-name)))
+    (t (error "Unable to find :use form in the defpackage form."))))
+
+
+(defun sly-import-symbol-from (symbol)
+  "Accepts "
+  (save-excursion
+   ;; First, will go to the package definition
+   (sly-goto-package-source-definition (sly-current-package))
+
+   (let* ((package (funcall sly-import-symbol-package-transform-function
+                            (sly-cl-symbol-package symbol)))
+          (simple-symbol (sly-cl-symbol-name symbol))
+          (import-exists (when package
+                           (sly-search-import-from package))))
+
+     ;; We only process symbols in fully qualified form like
+     ;; weblocks/request:get-parameter
+     (when package
+       (if import-exists
+           (let ((imported-symbols (mapcar #'sly-package-normalize-name
+                                           (sly-read-symbols))))
+             (unless (cl-member simple-symbol
+                                imported-symbols
+                                :test 'equalp)
+               ;; If symbol is not imported yet, then just
+               ;; add it to the end
+               ;; TODO: rename function
+               (sly-insert-export simple-symbol)))
+           ;; If there is no import from this package yet,
+           ;; then we'll add it right after the last :import-from
+           ;; or :use construction
+           (sly-package-create-new-import-from package
+                                               simple-symbol)))
+     ;; Always return symbol-without-package, because it is useful
+     ;; to replace symbol at point and change it from fully qualified
+     ;; form to a simple-form
+     simple-symbol)))
+
+
+(defun sly-import-symbol-at-point ()
+  "Takes a fully-qualified symbol at point, adds it to the defpackage
+   and replaces with a symbol name without a package part."
+  (interactive)
+  (let* ((bounds (sly-bounds-of-symbol-at-point))
+         (left-bound (car bounds))
+         (right-bound (cdr bounds)))
+    (when bounds
+      (save-excursion
+       ;; Put point to the beginning of the symbol to remove it at the end of the operation
+       (goto-char left-bound)
+       
+       (let* ((symbol-name
+                (buffer-substring-no-properties left-bound
+                                                right-bound))
+              (simple-symbol
+                (sly-import-symbol-from symbol-name)))
+         ;; If symbol was imported, then we need to replace
+         ;; fully qualified symbol name with simple one:
+         (when simple-symbol
+           ;; Now, remove symbol at the point from the buffer
+           (delete-region (point)
+                          (+ (point)
+                             (length symbol-name)))
```

Comment:
I think its clearer and more readable to use the `left-bound` and `right-bound` variables established earlier.

Reply:
Done.

Reply:
Oh, no, this does not work, because when we insert `:import-from` form, `left-bound` and `right-bound` become obsolete.

So, I've returned back `(point)` usage.

Reply:
Then use markers instead.